src/Microsoft.TestPlatform.ObjectModel/Client/Events/DiscoveryCompleteEventArgs.cs

```diff
@@ -42,4 +60,19 @@ public DiscoveryCompleteEventArgs(long totalTests, bool isAborted)
     /// Metrics
     /// </summary>
     public IDictionary<string, object> Metrics { get; set; }
+
+    /// <summary>
+    /// Gets the list of sources which were fully discovered.
+    /// </summary>
+    public IReadOnlyCollection<string> FullyDiscoveredSources { get; } = new List<string>();
```

Comment:
why are we creating default instances...are we expecting default creation or deserialization?

Reply:
I think it's because of the 2nd ctor that doesn't take the 3 lists. I am updating the implementation to make it more obvious.